flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -127,7 +129,10 @@ public ArchiveEventType getType() {
 	}
 
 	void start() {
-		executor.scheduleWithFixedDelay(fetcherTask, 0, refreshIntervalMillis, TimeUnit.MILLISECONDS);
+		final Runnable guardedTask = Runnables.withUncaughtExceptionHandler(
+			fetcherTask, FatalExitExceptionHandler.INSTANCE);
```

Comment:
Is this the right behaviour? When there is an exception now the JVM exits. Could there be more graceful handling?

Reply:
Not really, in our case we were swallowing OOMs, which leads to undefined state. History server code handles all exceptions, so this will be used only in case Error is thrown (bailing out is the most reasonable option here).

Reply:
Ok, and since it's a standalone component that should be ok.